metadata-db-app/src/cmr/metadata_db/data/oracle/concepts/variable.clj

```diff
@@ -23,3 +25,47 @@
         [cols values] (c/concept->common-insert-args concept)]
     [(concat cols ["provider_id" "user_id" "variable_name" "measurement"])
      (concat values [provider-id user-id variable-name measurement])]))
+
+(defn get-associations-for-variable-tombstone
+  "Returns the latest revisions of the variable associations for the given variable."
+  [db variable-tombstone]
+  (concepts/find-latest-concepts db
+                                 {:provider-id "CMR"}
+                                 {:concept-type :variable-association
+                                  ;; :variable name below refers to the table column
+                                  ;; named VARIABLE_NAME that holds the native-id
+                                  ;; value ... this comment will need to be updated
+                                  ;; once we move to using variable concept id.
+                                  :variable-name (:native-id variable-tombstone)}))
+
+(defn cascade-delete-variable-associations
```

Comment:
How about refactoring function and `cascade-delete-tag-associations` function into a single `cascade-delete-associations` and using the concept-type on the tombstone to determine which `get-associations-for-tombstone` (new multimethod replacing `get-associations-for-variable-tombstone`) to call?